src/librustc/ty/mod.rs

```diff
@@ -275,6 +275,22 @@ impl ImplOrTraitItemId {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+pub enum Visibility {
+    Public, //< Visible everywhere
+    Restricted(NodeId), //< Visible only in the given module
+    Private, //< Visible nowhere
```

Comment:
Private should always behave as `pub(self)`, so it may make sense to merge it into `Restricted(NodeId)`